AK/Optional.h

```diff
@@ -56,6 +56,21 @@ class alignas(T) [[nodiscard]] Optional {
         }
     }
 
+    ALWAYS_INLINE Optional(Optional& other)
```

Comment:
Is this needed?
`Optional::value()` is const, `Optional&` can be cast implicitly to `const Optional&`, and the copy ctor should be able to handle this use-case

Reply:
So it's not needed by itself, but it is needed after we add the `Optional(U& value)` overload.
Without `Optional(Optional& other)`, trying to copy-construct an `Optional` goes to the `Optional(U& value)` overload instead of `Optional(const Optional& value)` (because no implicit casts are needed?) and fails.

And the reason that `Optional(U& value)` is needed is to support the non const `HashMap::get()` overload, e.g to enable constructing `Optional<T*>` from an `OwnPtr<T>`. Without it, the only available way to construct the `Optional` is via `Optional(const U& value)`, which would mean the `OwnPtr` would have to be implicitly cast to `const OwnPtr`, and then it can't be turned into `T*`.